skbio/metadata/_mixin.py

```diff
@@ -406,3 +407,125 @@ def has_positional_metadata(self):
         """
         return (self._positional_metadata is not None and
                 len(self.positional_metadata.columns) > 0)
+
+
+class IntervalMetadataMixin(metaclass=abc.ABCMeta):
+    def __init__(self, interval_metadata=None):
+        raise NotImplementedError
+
+    def _init_(self, interval_metadata=None):
+        self.interval_metadata = interval_metadata
+
+    @property
+    @experimental(as_of="0.5.0-dev")
+    def interval_metadata(self):
+        '''``IntervalMetadata`` object containing info about interval features.
+
+        >>> from skbio import DNA
+        >>> from skbio.metadata import IntervalMetadata
+        >>> im = IntervalMetadata()
+        >>> intvl = im.add([(0, 3)], metadata={'gene': 'sagA'})
+        >>> seq = DNA('ACGT', interval_metadata=im)
+        >>> seq
+        DNA
+        --------------------------
+        Interval metadata:
+            1 interval features
+        Stats:
+            length: 4
+            has gaps: False
+            has degenerates: False
+            has definites: True
+            GC-content: 50.00%
+        --------------------------
+        0 ACGT
+        '''
+        if self._interval_metadata is None:
+            # Not using setter to avoid copy.
+            self._interval_metadata = IntervalMetadata()
```

Comment:
"axis length" makes sense for the object this mixin is applied to, since some objects have more than one axis (e.g. for `TabularMSA`, it could be attached to rows or columns. `Sequence` is less interesting.). For `IntervalMetadata` though, it really is a closed interval `[0, N]`. `N` is what we're trying to name here (which will be the constructor parameter). Using "axis" in `IntervalMetadata` doesn't seem as clear to me, that's why I was thinking something like the `upper_bound` of the interval would make more sense. Does that distinction make sense, or is "axis" really applicable to `IntervalMetadata` standalone? Note that this is similar to how positional metadata mixin defines `_positional_metadata_axis_len_`, and the `pd.DataFrame`'s "length" is the number of rows (the two don't directly correspond, since a data frame has two axes).